pyanaconda/payload/dnf/payload.py

```diff
@@ -1257,6 +1263,54 @@ def pre_install(self):
         elif groupid:
             log.warning("Platform group %s not available.", groupid)
 
+    def _set_up_fips(self):
+        """Set up FIPS in the target system.
+
+        Copy the crypto policy from the installation environment
+        to target system the before the package installation. The
+        RPM scriptlets needs to be executed in the FIPS mode if
+        there is fips=1 on the kernel cmdline.
+
+        FIXME: Move the FIPS support to the Security module.
+         """
+        log.debug("Copying the crypto policy.")
+
+        if not self._check_fips():
+            raise SecurityInstallationError(
+                "FIPS is not correctly set up "
+                "in the installation environment."
+            )
+
+        src = "/etc/crypto-policies/back-ends/"
+        dst = os.path.normpath(conf.target.system_root + src)
+
+        util.mkdirChain(src)
+        shutil.copytree(src, dst, symlinks=True)
+
+        # Log the directory content of the target system.
+        util.execWithRedirect("/bin/ls", ["-l", dst])
+
+    def _check_fips(self):
+        """Check FIPS in the installation system."""
+        config_path = "/etc/crypto-policies/config"
```

Comment:
These paths should be moved to constants but we can do that later. No need to do it right now.